magento/magento2
# 11946: Layer navigation showing wrong product count

### Description
Added check if need to show out of stock products.

### Fixed Issues (if relevant)
1. magento/magento2#11946: Layer navigation showing wrong product count

### Manual testing scenarios
1. Create simple product and attach it to category.
2. Find simple product on frontend catalog(Layer navigation is showing all visibility product count).
3. Enable "Display Out of Stock Products" (Stores->Configuration->Catalog->Inventory)
4. Set a simple product "out of stock"
5. Go to category with that product
5. Layer navigation should show the same  product count as in step 2.

### Contribution checklist
